packages/angular_devkit/core/src/logger/logger.ts

```diff
@@ -161,7 +161,8 @@ export class Logger extends Observable<LogEntry> implements LoggerApi {
       arguments as unknown as Parameters<Observable<LogEntry>['subscribe']>,
     );
   }
-  override forEach(next: (value: LogEntry) => void, PromiseCtor?: typeof Promise): Promise<void> {
-    return this._observable.forEach(next, PromiseCtor);
+
+  override forEach(next: (value: LogEntry) => void, promiseCtor?: typeof Promise): Promise<void> {
+    return this._observable.forEach(next, promiseCtor as PromiseConstructor);
```

Comment:
The casting here is redundant. 

```suggestion
    return this._observable.forEach(next, promiseCtor);
```

Reply:
Actually, it is an optional value, and when `strictNullChecks` is enabled, compiler throws an error for this line when nothing is done. I initially added a non-null assertion `!` but linter did not like it, so I added this cast.
I can take an alternative approach if you have a suggestion.

Reply:
Done, thank you for your guidance. Please take a look.

Reply:
I am a bit confused by your statement as strictNullChecks are already enabled and the build currently succeeds.

Reply:
Let’s do that.I am good with the other changes though. So if you want to revert this I will LGTM.

Reply:
I reverted the cast, but I kept the variable naming change, as it was confusing. `PromiseCtor` looked more like a type rather than a local variable. Is that OK?

Reply:
Yes, I totally agree with you